tests/unit/expire.tcl

```diff
@@ -601,4 +601,117 @@ start_server {tags {"expire"}} {
            {del foo}
         }
     } {} {needs:repl}
+
+    test {EXPIRE with NX option on a key with ttl} {
+        r SET foo bar EX 100000
+        assert_equal [r EXPIRE foo 200000 NX] 0
+        assert_range [r TTL foo] 50000 100000
+    } {}
+
+    test {EXPIRE with NX option on a key without ttl} {
+        r SET foo bar
+        assert_equal [r EXPIRE foo 200000 NX] 1
+        assert_range [r TTL foo] 100000 200000
+    } {}
+
+    test {PEXPIRE with NX option on a key with ttl} {
+        r SET foo bar EX 100000
+        assert_equal [r PEXPIRE foo 200000 NX] 0
+        assert_range [r TTL foo] 50000 100000
+    } {}
+
+    test {PEXPIRE with NX option on a key without ttl} {
+        r SET foo bar
+        assert_equal [r PEXPIRE foo 200000 NX] 1
+        assert_range [r TTL foo] 100 200
+    } {}
+
+    test {EXPIRE with XX option on a key with ttl} {
+        r SET foo bar EX 100000
+        assert_equal [r EXPIRE foo 200000 XX] 1
+        assert_range [r TTL foo] 100000 200000
+    } {}
+
+    test {EXPIRE with XX option on a key without ttl} {
+        r SET foo bar
+        assert_equal [r EXPIRE foo 200000 XX] 0
+        assert_equal [r TTL foo] -1
+    } {}
+
+    test {EXPIRE with GT option} {
+        r SET foo bar EX 100000
+        assert_equal [r EXPIRE foo 200000 GT] 1
+        assert_range [r TTL foo] 100000 200000
+    } {}
+
+    test {EXPIRE with GT option} {
+        r SET foo bar EX 200000
+        assert_equal [r EXPIRE foo 100000 GT] 0
+        assert_range [r TTL foo] 100000 200000
+    } {}
+
+    test {EXPIRE with LT option} {
+        r SET foo bar EX 100000
+        assert_equal [r EXPIRE foo 200000 LT] 0
+        assert_range [r TTL foo] 50000 100000
+    } {}
+
+    test {EXPIRE with LT option} {
```

Comment:
again, let's rename both tests, one saying `low ttl` and the other saying `high ttl`